Library/EnvelopeFactory.ts

```diff
@@ -135,10 +137,27 @@ class EnvelopeFactory {
         }
     }
 
+    private static truncateProperties(telemetry: Telemetry) {
+        if (telemetry.properties) {
+            let properties: {[key: string]: any} = {};
```

Comment:
You need to ensure this code doesn't throw an error with weird properties, this is code executed for all telemetry events, we need to be extra cautious here

Reply:
Added some extra error handling here. The call to serialize the values was already handled by the util function, but makes sense to wrap the whole method in case of seriously unexpected input.